src/core/document.js

```diff
@@ -578,30 +578,53 @@ class Page {
     return tree;
   }
 
-  getAnnotationsData(intent) {
-    return this._parsedAnnotations.then(function (annotations) {
-      const annotationsData = [];
+  async getAnnotationsData(handler, task, intent) {
+    const annotations = await this._parsedAnnotations;
+    if (annotations.length === 0) {
+      return [];
+    }
 
-      if (annotations.length === 0) {
-        return annotationsData;
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: this.pageIndex,
+      idFactory: this._localIdFactory,
+      fontCache: this.fontCache,
+      builtInCMapCache: this.builtInCMapCache,
+      standardFontDataCache: this.standardFontDataCache,
+      globalImageCache: this.globalImageCache,
+      options: this.evaluatorOptions,
+    });
+    const textContentPromises = [];
+    const annotationsData = [];
```

Comment:
My final suggestion would be that we don't create a `PartialEvaluator`-instance unconditionally, since most pages won't actually need that:
```suggestion
    const annotationsData = [],
      textContentPromises = [];
    let partialEvaluator;
```